### ISSUE TITLE
Offline page should periodically test for restored connection

### ISSUE THREAD

**Issue by TalAter** [TalAter/UpUp#42]

**Comment by hye** [comment]
:+1: 
and  maybe `navigator.onLine` should check first

**Comment by TalAter** [comment]
Interesting, I didn't know this one. Thank you :+1: 

> The property sends updates whenever the browser's ability to connect to the network changes. The update occurs when the user follows links or when a script requests a remote page.
>
> In Chrome and Safari, if the browser is not able to connect to a local area network (LAN) or a router, it is offline; all other conditions return true. So while you can assume that the browser is offline when it returns a false value, you cannot assume that a true value necessarily means that the browser can access the internet.
> - https://developer.mozilla.org/en-US/docs/Web/API/NavigatorOnLine/onLine

Might not be good enough… If a user just has the page sitting in the browser, waiting for the connection to go back online, `navigator.onLine` won't change… not until the browser tries to access the network. Perhaps we need to initiate a network request periodically (need to make sure this is only done when a page in the scope is shown).

**Comment by ksnyde** [comment]
I think the `onLine` indicator would be nice but I think it's implementation is inconsistent across the different browsers. I think periodic network polling is unfortunately the only reliable way of testing this (see offline.js for a popular solution). This "cost" however is pretty minimal and since it may in fact be beneficial to retry some of the cached content from time to time you could kill two birds with one stone (aka, check the network by checking for updates to content of existing cache content).

**Comment by TalAter** [comment]
@ksnyde I agree, and the checked cached content with the same stone is a great idea!
